core/arch/arm/include/mm/core_memprot.h

```diff
@@ -45,14 +45,13 @@
 
 /* memory atttributes */
 enum buf_is_attr {
-	CORE_MEM_SEC,
+	CORE_MEM_CACHED,
+	CORE_MEM_EXTRAM,
+	CORE_MEM_NSEC_SHM,
 	CORE_MEM_NON_SEC,
+	CORE_MEM_SEC,
 	CORE_MEM_TEE_RAM,
 	CORE_MEM_TA_RAM,
-	CORE_MEM_NSEC_SHM,
-	CORE_MEM_EXTRAM,
-	CORE_MEM_INTRAM,
-	CORE_MEM_CACHED,
```

Comment:
I am in favor of trying to better detect misuses of `enum buf_is_attr` vs. `enum teecore_memtypes`, but I think this is a bit fragile. How about using different ranges and adding a few `asserts()`? Like this more or less:

```C
enum buf_is_attr {
        CORE_MEM_FIRST = 20, /* to detect misuse of teecore_memtypes */
        CORE_MEM_SEC,
        CORE_MEM_NON_SEC,
        /* ... */
};

enum teecore_memtypes {
        MEM_AREA_NOTYPE = 0,
        MEM_AREA_TEE_RAM,
        /* ... */
        MEM_AREA_TYPE_LAST
};

void foo(enum buf_is_attr attr)
{
        assert(attr > CORE_MEM_FIRST);
        /* ... */
}


void bar(enum teecore_memtypes type)
{
        assert(type < MEM_AREA_TYPE_LAST);
        /* ... */
}
```